src/vm/functions/special.rs

```diff
@@ -48,38 +48,25 @@ fn parse_pox_stacking_result(
 /// Handle special cases when calling into the PoX API contract
 fn handle_pox_api_contract_call(
     global_context: &mut GlobalContext,
-    sender_opt: Option<&PrincipalData>,
+    _sender_opt: Option<&PrincipalData>,
     function_name: &str,
     value: &Value,
 ) -> Result<()> {
-    if function_name == "stack-stx" {
+    if function_name == "stack-stx" || function_name == "delegator-stack-stx" {
         debug!(
             "Handle special-case contract-call to {:?} {} (which returned {:?})",
             boot_code_id("pox"),
             function_name,
             value
         );
 
-        // sender is required
-        let sender = match sender_opt {
-            None => {
-                return Err(RuntimeErrorType::NoSenderInContext.into());
-            }
-            Some(sender) => (*sender).clone(),
-        };
-
         match parse_pox_stacking_result(value) {
             Ok((stacker, locked_amount, unlock_height)) => {
-                assert_eq!(
-                    stacker, sender,
-                    "BUG: tx-sender is not contract-call origin!"
-                );
-
                 // if this fails, then there's a bug in the contract (since it already does
                 // the necessary checks)
                 match StacksChainState::pox_lock(
                     &mut global_context.database,
-                    &sender,
+                    &stacker,
```

Comment:
Just to help my own understanding -- this is the `stacker` now, not the `sender`, because the `stacker` will now reflect either the principal that owns the STX (i.e. `tx-sender`), or the specific delegated principal that the STX-owner set.  Is my understanding correct?

Reply:
Yep